test/python/pulse/test_transforms.py

```diff
@@ -401,12 +407,286 @@ def test_multiple_schedules(self):
+        reference.insert(0, instructions.Delay(3, d0), mutate=True)
+        reference.insert(8, instructions.Delay(7, d0), mutate=True)
+        # d1
+        reference.insert(3, instructions.Delay(5, d1), mutate=True)
+
+        self.assertEqual(schedule, reference)
+
+    def test_align_sequential_with_barrier(self):
+        """Test sequential alignment with a barrier."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+
+        schedule = pulse.Schedule()
+        schedule = schedule.insert(1, instructions.Delay(3, d0), mutate=True)
+        schedule.append(directives.RelativeBarrier(d0, d1), mutate=True)
+        schedule = schedule.insert(4, instructions.Delay(5, d1), mutate=True)
+        schedule = schedule.insert(12, instructions.Delay(7, d0), mutate=True)
+
+        schedule = transforms.remove_directives(
+            transforms.align_sequential(schedule))
+
+        reference = pulse.Schedule()
+        # d0
+        reference.insert(0, instructions.Delay(3, d0), mutate=True)
+        reference.insert(8, instructions.Delay(7, d0), mutate=True)
+        # d1
+        reference.insert(3, instructions.Delay(5, d1), mutate=True)
+
+        self.assertEqual(schedule, reference)
+
+
+class TestAlignLeft(QiskitTestCase):
+    """Test left alignment transform."""
+
+    def test_align_right(self):
+        """Test left alignment without a barrier."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+        d2 = pulse.DriveChannel(2)
+
+        schedule = pulse.Schedule()
+        schedule = schedule.insert(1, instructions.Delay(3, d0), mutate=True)
+        schedule = schedule.insert(17, instructions.Delay(11, d2), mutate=True)
+
+        sched_grouped = pulse.Schedule()
+        sched_grouped += instructions.Delay(5, d1)
+        sched_grouped += instructions.Delay(7, d0)
+        schedule.append(sched_grouped, mutate=True)
+
+        schedule = transforms.align_left(schedule)
+        reference = pulse.Schedule()
+        # d0
+        reference += instructions.Delay(3, d0)
+        reference += instructions.Delay(7, d0)
+        # d1
+        reference = reference.insert(3, instructions.Delay(5, d1))
+        # d2
+        reference += instructions.Delay(11, d2)
+
+        self.assertEqual(schedule, reference)
+
+    def test_align_left_with_barrier(self):
+        """Test left alignment with a barrier."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+        d2 = pulse.DriveChannel(2)
+
+        schedule = pulse.Schedule()
+        schedule = schedule.insert(1, instructions.Delay(3, d0), mutate=True)
+        schedule.append(directives.RelativeBarrier(d0, d1, d2), mutate=True)
+        schedule = schedule.insert(17, instructions.Delay(11, d2), mutate=True)
+
+        sched_grouped = pulse.Schedule()
+        sched_grouped += instructions.Delay(5, d1)
+        sched_grouped += instructions.Delay(7, d0)
+        schedule.append(sched_grouped, mutate=True)
```

Comment:
Does `_grouped` have anything to do with `pulse.group`?

Reply:
In a sense it does, since `pulse.group` is just a trivial embedding of instructions in a schedule so that a schedule can be embedded in a schedule. This is what I'm trying to demonstrate here.